android/src/main/kotlin/com/onegini/mobile/sdk/flutter/helpers/SdkError.kt

```diff
@@ -0,0 +1,128 @@
+package com.onegini.mobile.sdk.flutter.helpers
+
+import com.onegini.mobile.sdk.flutter.OneWelcomeWrapperErrors.*
+import com.onegini.mobile.sdk.flutter.OneWelcomeWrapperErrors
+import com.onegini.mobile.sdk.flutter.constants.Constants.Companion.RESPONSE_BODY
+import com.onegini.mobile.sdk.flutter.constants.Constants.Companion.RESPONSE_HEADERS
+import com.onegini.mobile.sdk.flutter.constants.Constants.Companion.RESPONSE_STATUS_CODE
+import com.onegini.mobile.sdk.flutter.constants.Constants.Companion.RESPONSE_URL
+import io.flutter.plugin.common.MethodChannel
+import okhttp3.Response
+
+class SdkError {
+    private val code: Int
+    private val message: String
+    private val details: MutableMap<String, Any> = mutableMapOf()
+
+    // Only error codes
+    constructor(
+        code: Int,
+        message: String?,
+    ) {
+        this.code = code
+        this.message = when(message) {
```

Comment:
You don't need those `when` statements here. If it's a simple null check, you can use an elvis operator `:?` `this.message = message ?: GENERIC_ERROR.message`
follow the pattern for all the cases in this class.

Reply:
thanks for the tip, reworked and will make elvis proud from now on